ui/src/cloud/reducers/orgsettings.ts

```diff
@@ -0,0 +1,26 @@
+import {produce} from 'immer'
+
+import {Actions, ActionTypes} from 'src/cloud/actions/orgsettings'
+import {OrgSetting} from 'src/types'
+
+import {PAID_ORG_HIDE_UPGRADE_SETTING} from 'src/cloud/constants'
+
+export interface OrgSettingsState {
+  settings: OrgSetting[]
+}
+
+export const defaultState: OrgSettingsState = {
+  settings: [PAID_ORG_HIDE_UPGRADE_SETTING],
+}
+
+export const orgSettingsReducer = (
+  state = defaultState,
+  action: Actions
+): OrgSettingsState =>
+  produce(state, draftState => {
+    if (action.type === ActionTypes.SetOrgSettings) {
+      const {orgSettings} = action.payload
+      draftState.settings = orgSettings
```

Comment:
stylistic nit: why not compact this to a single line? grepping for `orgSettings` will still find this line, so no information or context is lost. 

```suggestion
      draftState.settings = action.payload.orgSettings
```

Reply:
i put that in there to make sure you were paying attention 😁